### ISSUE TITLE
[iOS] WKWebView auto-scroll to top after push screen from root StackNavigator

### ISSUE THREAD

**Issue by malonguwa** [react-native-community/react-native-webview#201]
@Titozzz 
I was using 2.5.0, and I see the re-create issue then I talk to the @NeilChan and come up with a sultion (https://github.com/react-native-community/react-native-webview/pull/143), in 2.5.0, I just go updated WKWebView.m by following this PR, everything works well, but today after I update to the latest version, the issue is happening again, but this time seems related to the react navigation, I have a structure like this

```
StackNavigator(root)
-> TabNavigator
      -> StackNavigator
            ->  My WebView is in this StackNavigator
-> ModalScreen

```
In my webView when I click my HTML button, and then I will intercept the customised URL sent from my HTML button, and use the root StackNavigator to push a ModalScreen screen.

But at the moment when the screen is being pushed, the webView being re-create. I can not say there is react navigation's bug, due to with webView v2.5.0 and updated the WKWebView.m a little bit does not have the issue I have now, so there must something changed after 2.5.0

**Comment by NeilChan** [comment]
@malonguwa 
It think it maybe affected by react navigation's render mechanism. The react navigation seems didn't work like the native navigation. The original webview maybe removed from superview after pushed a new screen. As the native navigation, it won't be.

You can remove the method `- (void)removeFromSuperview()` in `RNCWKWebView.m` temporary and test again. 
But it maybe cause a memory leak.

Now I have no better idea to fix this situation. I will find a better way to fix it and submit a PR later.

**Comment by malonguwa** [comment]
@NeilChan Hi Neil, many thanks to look into the issue again! Yes... you are right.. may have memory leak... Hope you can find a better way to fix this... Good luck man!

**Comment by NeilChan** [comment]
@malonguwa 
If you remove the method - (void)removeFromSuperview() in RNCWKWebView.m can fix the problem ?

**Comment by malonguwa** [comment]
@NeilChan Hi sorry for the late reply, after I remove the removeFromSuperview() in RNCWKWebView.m , the re-create issue still there if I push screen from the root stackNavigator.

**Comment by malonguwa** [comment]
@NeilChan interesting thing is if I use your fix for RNCWKWebView.m in react-native-webview: 2.5.0, I do not see this issue...  not sure what has been changed after 2.5.0

**Comment by malonguwa** [comment]
@NeilChan @Titozzz 

I find the code which caused this issue, after I commented off the following code, my issue is solved
```
  // Workaround for a keyboard dismissal bug present in iOS 12
  // https://openradar.appspot.com/radar?id=5018321736957952
  if (@available(iOS 12.0, *)) {
    [[NSNotificationCenter defaultCenter]
      addObserver:self
      selector:@selector(keyboardWillHide)
      name:UIKeyboardWillHideNotification object:nil];
    [[NSNotificationCenter defaultCenter]
      addObserver:self
      selector:@selector(keyboardWillShow)
      name:UIKeyboardWillShowNotification object:nil];
  }
```

**Comment by jamonholmgren** [comment]
@malonguwa Thanks, that's helpful. We're experiencing something similar with a client project so we'll take a look at this too.

**Comment by ryanlntn** [comment]
@jamonholmgren This is definitely the same bug we're experiencing. Commenting that block out resolves the scroll to top on select input blur.

**Comment by jamonholmgren** [comment]
@malonguwa @ryanlntn Let's revert that commit and try again with that particular bug later.

**Comment by ryanlntn** [comment]
On it.

**Comment by NeilChan** [comment]
@malonguwa As you said, I don’t think it has been re-created. The code just cause the scroll offset changed.

**Comment by iamtommcc** [comment]
@malonguwa @jamonholmgren Sadly I wouldn't have any idea as to why #134 would cause this issue. Does the problem occur on all versions of iOS & Xcode?

Quite a dicey situation. The original bug my PR fixed is pretty bad, but this issue new introduced is also pretty bad so a revert was probably a good idea 😅.

**Comment by jamonholmgren** [comment]
This has been fixed in #296.
